packages/deployed-backend-client/src/deployed_backend_client_factory.ts

```diff
@@ -26,14 +26,15 @@ export enum ApiAuthType {
   AMAZON_COGNITO_USER_POOLS = 'AMAZON_COGNITO_USER_POOLS',
 }
 
-export type SandboxMetadata = {
+export type ListBackendsMetadata = {
   name: string;
   lastUpdated: Date | undefined;
   status: BackendDeploymentStatus;
   backendId: BackendIdentifier | undefined;
 };
 
-export type ListSandboxesRequest = {
+export type ListBackendsRequest = {
+  deploymentType: string;
   nextToken?: string;
```

Comment:
We probably need to remove this and return some sort of Iterable object with a `.next(token?: string): BackendMetadata[]` method to get the next batch of results. Reason being, what happens when someone passes in a nextToken but with different filter criteria?

Reply:
We could also consider using [yield](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/yield) to return an Iterable that can keep track of the nextToken internally